src/super_gradients/training/utils/quantization/core.py

```diff
@@ -0,0 +1,140 @@
+
+    # Edge-cases
+    if 'bias' in float_instance_state:
+        if float_instance_state['bias'] is None:  # None is the state when bias=False in torch.nn
+            float_instance_state['bias'] = False
+        elif not isinstance(float_instance_state['bias'], bool):  # Tensor is the state when bias=True in torch.nn
+            float_instance_state['bias'] = True
+        # in case bias is a boolean - we don't do anything, so it is taken as-is, either True or False
+    return float_instance_state
+
+
+def _from_float(cls, float_instance, **kwargs):
+    init_params = _extract_init_args(cls, float_instance)
+    init_params.update(**kwargs)
+    return cls(**init_params)
+
+
+class SGQuantMixin(nn.Module):
+    """
+    A base class for user custom Quantized classes.
+    Every Quantized class must inherit this mixin, which adds `from_float` class-method.
+    NOTES:
+        * the Quantized class may also inherit from the native `QuantMixin` or `QuantInputMixin`
+        * quant descriptors (for inputs and weights) will be passed as `kwargs`. The module may ignore them if they are
+          not necessary
+        * the default implementation of `from_float` is inspecting the __init__ args, and searching for corresponding
+          properties from the float instance that is passed as argument, e.g., for `__init__(self, a)`
+          the mechanism will look for `float_instance.a` and pass that value to the `__init__` method
+    """
+
+    @classmethod
+    def from_float(cls, float_instance, **kwargs):
+        return _from_float(cls, float_instance, **kwargs)
+
+
+class SkipQuantization(nn.Module):
+    """
+    This class wraps a float module instance, and defines that this instance will not be converted to quantized version
+
+    Example:
+        self.my_block = SkipQuantization(MyBlock(4, n_classes))
+    """
+
+    def __init__(self, module: nn.Module) -> None:
+        super().__init__()
+        self.float_module = module
+        self.forward = module.forward
+
+
+class QuantizedMapping(nn.Module):
+    """
+    This class wraps a float module instance, and defines a mapping from this instance to the corresponding quantized
+    class, with relevant quant descriptors.
+
+    Example:
+        self.my_block = QuantizedMapping(float_module=MyBlock(4, n_classes), quantized_type=MyQuantizedBlock)
+    """
+
+    def __init__(self, *, float_module: nn.Module,
+                 quantized_type: Union[Type[QuantMixin], Type[QuantInputMixin], Type[SGQuantMixin]],
+                 input_quant_descriptor: QuantDescriptor = None,
+                 weights_quant_descriptor: QuantDescriptor = None) -> None:
+        super().__init__()
+        self.float_module = float_module
+        self.quantized_type = quantized_type
+        self.input_quant_descriptor = input_quant_descriptor,
+        self.weights_quant_descriptor = weights_quant_descriptor,
+        self.forward = float_module.forward
```

Comment:
Can we rewrite this inside this modules forward? It's a bit hacky..

Reply:
Sure we can, however, I want to be invariant for those weird `forward`s that accept more than one argument.
We can use `*args` though... whatever you like